src/plugins/tonegen/ausrv.c

```diff
@@ -0,0 +1,332 @@
+/*************************************************************************
+This file is part of tone-generator
+
+Copyright (C) 2010 Nokia Corporation.
+
+This library is free software; you can redistribute
+it and/or modify it under the terms of the GNU Lesser General Public
+License as published by the Free Software Foundation
+version 2.1 of the License.
+
+This library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public
+License along with this library; if not, write to the Free Software
+Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301
+USA.
+*************************************************************************/
+
+#define _GNU_SOURCE
+
+#include <stdlib.h>
+#include <string.h>
+#include <errno.h>
+
+#include <ngf/log.h>
+#include <trace/trace.h>
+
+#include "stream.h"
+#include "ausrv.h"
+
+#if PA_API_VERSION < 9
+#error Invalid PulseAudio API version
+#endif
+
+#define DISCONNECTED    0
+#define CONNECTED       1
+
+#define DEFAULT_SERVER  "default Pulse Audio"
+#define CONNECT_DELAY   10                              /* in seconds */
+
+#define TRACE(f, args...) trace_write(trctx, trflags, trkeys, f, ##args)
+#define LOG_CAT "tonegen-ausrv: "
+
+static void set_connection_status(struct ausrv *, int);
```

Comment:
This could use a `bool` (from `stdbool.h`) instead of the two defines `DISCONNECTED` and `CONNECTED`.